internal/bed/read.go

```diff
@@ -0,0 +1,84 @@
+package bed
+
+import (
+	"bufio"
+	"fmt"
+	"io"
+	"os"
+	"regexp"
+	"strconv"
+	"strings"
+)
+
+// Opening and reading the bed file
+func (bf *Bedfile) Read() error {
+	file, err := os.Open(bf.Input)
+	if err != nil {
+		return err
+	}
+	defer file.Close()
+	return bf.read(file)
+}
+
+// Reading the bed file
+func (bf *Bedfile) read(file io.Reader) error {
+	var err error
+	var expectedNrOfCols int
+
+	minNrCols := 3
+	headerPattern := regexp.MustCompile(`^(browser|track|#)`)
+	strandPattern := regexp.MustCompile(`^(\.|\+|-|\+1|-1|1)$`)
+
+	lineNr := 0
+	scanner := bufio.NewScanner(file)
+	for scanner.Scan() {
+		var l Line
+		lineNr++
```

Comment:
I first was like "why are you indexing the lines" but then I saw that you are using them to log the line for error messages. Smart 💡